api/dbadapters/sqldatawarehouse.ts

```diff
@@ -128,28 +125,54 @@ export class SQLDataWarehouseDBAdapter implements IDbAdapter {
   }
 
   public async tables(): Promise<dataform.ITarget[]> {
-    const {
-      rows
-    } = await this.execute(
-      `select ${TABLE_SCHEMA_COL_NAME}, ${TABLE_NAME_COL_NAME} from ${INFORMATION_SCHEMA_SCHEMA_NAME}.tables`,
-      { rowLimit: 10000 }
+    const { rows } = await this.execute(
+      `select table_schema, table_name from information_schema.tables`,
+      {
+        rowLimit: 10000
+      }
     );
     return rows.map(row => ({
-      schema: row[TABLE_SCHEMA_COL_NAME],
-      name: row[TABLE_NAME_COL_NAME]
+      schema: row.table_schema,
+      name: row.table_name
     }));
   }
 
+  public async search(searchText: string): Promise<dataform.ITableMetadata[]> {
+    const results = await this.execute(
+      `select tables.table_schema as table_schema, tables.table_name as table_name
+       from information_schema.tables as tables
+       left join information_schema.columns as columns on tables.table_schema = columns.table_schema and tables.table_name = columns.table_name
+       where tables.table_schema like @searchText or tables.table_name like @searchText or columns.column_name like @searchText
+       group by tables.table_schema, tables.table_name`,
+      {
+        params: {
+          searchText: `%${searchText}%`
+        },
+        rowLimit: 100
+      }
+    );
+    return await Promise.all(
+      results.rows.map(row =>
+        this.table({
+          schema: row.table_schema,
+          name: row.table_name
+        })
+      )
+    );
+  }
+
   public async table(target: dataform.ITarget): Promise<dataform.ITableMetadata> {
     const [tableData, columnData] = await Promise.all([
       this.execute(
-        `select ${TABLE_TYPE_COL_NAME} from ${INFORMATION_SCHEMA_SCHEMA_NAME}.tables
-          where ${TABLE_SCHEMA_COL_NAME} = '${target.schema}' AND ${TABLE_NAME_COL_NAME} = '${target.name}'`
+        `select table_type from information_schema.tables
+         where table_schema = @schema AND table_name = @name`,
+        { params: target }
       ),
       this.execute(
-        `select ${COLUMN_NAME_COL_NAME}, ${DATA_TYPE_COL_NAME}, ${IS_NULLABLE_COL_NAME}
-       from ${INFORMATION_SCHEMA_SCHEMA_NAME}.columns
-       where ${TABLE_SCHEMA_COL_NAME} = '${target.schema}' AND ${TABLE_NAME_COL_NAME} = '${target.name}'`
+        `select column_name, data_type, is_nullable
+         from information_schema.columns
+         where table_schema = @schema AND table_name = @name`,
+        { params: target }
```

Comment:
nit: I'd suggest rebuilding these maps, as I'm sure we'll one day change those protos and maybe not spot this